packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -272,6 +273,7 @@ function WrappedDay<TDate extends unknown>({
       isAnimating: isMonthSwitchingAnimating,
       // it is used in date range dragging logic by accessing `dataset.timestamp`
       'data-timestamp': utils.toJsDate(day).valueOf(),
+      ...other,
```

Comment:
If we spread, we can remove `onDaySelect`, `onKeyDown`, `onBlur` and `onFocus` for the explicitly defined props.